Description: Fix XFIXES selection handling (copy and paste via middle mouse button).
Abstract:
 When nxagent has the XFIXES extension enabled copy and
 paste from outside applications to applications within the session
 that rely on XFixesSelectSelectionInput (e.g. qt applications like
 konsole) did never receive any notifications because the nxagent did
 not register itself at the real X server to receive them. Fixes Bug
 #585.
Author: Ulrich Sibiller <uli42@gmx.de>

--- a/nx-X11/programs/Xserver/hw/nxagent/Clipboard.c
+++ b/nx-X11/programs/Xserver/hw/nxagent/Clipboard.c
@@ -1556,10 +1556,14 @@ int nxagentInitClipboard(WindowPtr pWin)
     fprintf(stderr, "nxagentInitClipboard: Registering for XFixesSelectionNotify events.\n");
     #endif
 
-    XFixesSelectSelectionInput(nxagentDisplay, iWindow, nxagentClipboardAtom,
-                               XFixesSetSelectionOwnerNotifyMask |
-                               XFixesSelectionWindowDestroyNotifyMask |
-                               XFixesSelectionClientCloseNotifyMask);
+    for (i = 0; i < nxagentMaxSelections; i++)
+    {
+      XFixesSelectSelectionInput(nxagentDisplay, iWindow,
+                                 lastSelectionOwner[i].selection,
+                                 XFixesSetSelectionOwnerNotifyMask |
+                                 XFixesSelectionWindowDestroyNotifyMask |
+                                 XFixesSelectionClientCloseNotifyMask);
+    }
 
     nxagentXFixesInfo.Initialized = 1;
   }
